libr/cons/grep.c

```diff
@@ -597,7 +597,8 @@ R_API void r_cons_grepbuf(void) {
 			break;
 		}
 		l = p - in;
-		if (l > 0) {
+		if (((grep->range_line == 2 || *grep->str) && l > 0) ||
```

Comment:
extract this big check in a  function and reuse it

Reply:
i think a function is a bit heavy, but i've simplified the check expression f766a50b7738525d45649eb9d5e5530c0b3c18b5